### PR TITLE
My Profile create

### PR DESCRIPTION
**Pull request by Nazarii-Makohin**
1. Created simple <My Profile> page for each User
2. added new Model and Controller <Profile>
3. fixed css bugs with Active_Admin
4. Added 2 migrations for profile and user to have same user_id
5. New way tyo make a  registration and log in

### CODE REVIEW

File: app/assets/stylesheets/active_admin.scss [hunk 1]
```diff
@@ -10,7 +10,9 @@
 // Active Admin's got SASS!
 @import "active_admin/mixins";
 @import "active_admin/base";
+a:link { text-decoration:none; }
```
Comment (olegsobchuk) [inline app/assets/stylesheets/active_admin.scss:13]: add new line to separate styles from `@import` block

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -10,6 +10,10 @@
...
- *= require_tree .
- *= require_self
- */
+*=require_tree .
+
+*= stub "active_admin"
+
+*= require_self
+
+ */
```
Comment (olegsobchuk) [inline app/assets/stylesheets/application.css:19]: add new empty line in the end of file

Comment (olegsobchuk) [reply app/assets/stylesheets/application.css:19]: you can set up your IDE to do it automatically

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -2,6 +2,6 @@ class PagesController < ApplicationController
   def home
   end
   def about
-    
+
   end
-end
+end
```
Comment (olegsobchuk) [inline app/controllers/pages_controller.rb:7]: the same with empty line

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -2,6 +2,6 @@ class PagesController < ApplicationController
   def home
   end
   def about
-    
+
```
Comment (olegsobchuk) [inline app/controllers/pages_controller.rb:5]: if you have empty methods, write it in one string
bad
```
def about
end
```
good
```
def about; end
```

File: app/controllers/profiles_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+class ProfilesController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by!(params[:id])
```
Comment (olegsobchuk) [inline app/controllers/profiles_controller.rb:7]: suppose it doesn't work `find_by` requires key-val params

Comment (olegsobchuk) [reply app/controllers/profiles_controller.rb:7]: why not `User.find(params[:id])` ???

File: app/controllers/profiles_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def new
+  @user = User.new
+end
+
+  def edit
+    @user = User.find_by!(id: params[:id])
```
Comment (olegsobchuk) [inline app/controllers/profiles_controller.rb:15]: DRY ???
use `before_action` for this

File: app/controllers/profiles_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to profiles_show_url, notice: 'User was successfully created.'
```
Comment (olegsobchuk) [inline app/controllers/profiles_controller.rb:22]: `_url` ???? `_path`

File: app/controllers/profiles_controller.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    if @user.update(user_params)
+      redirect_to profiles_show_url
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @event = User.find_by!(params[:id])
+    @event.destroy
```
Comment (olegsobchuk) [inline app/controllers/profiles_controller.rb:40]: `User.destroy(params[:id])`

File: app/helpers/profiles_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module ProfilesHelper
```
Comment (olegsobchuk) [inline app/helpers/profiles_helper.rb:1]: remove this file if you don't need it

File: app/models/profile.rb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+class Profile < ApplicationRecord
+
```
Comment (olegsobchuk) [inline app/models/profile.rb:2]: remove empty line after `class` definition

File: app/models/profile.rb [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :user
+  has_one_attached :avatar
+
+  validates :email,   presence: true
```
Comment (olegsobchuk) [inline app/models/profile.rb:6]: you can use one string for this
```
validates :email, :user_id, presence: true
```

File: app/models/user.rb [hunk 12]
```diff
@@ -1,10 +1,17 @@
 class User < ApplicationRecord
+
+  has_one :profile, dependent: :destroy
+
+  has_one_attached :avatar
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (olegsobchuk) [inline app/models/user.rb:7]: remove comments

File: app/views/events/_form.html.erb [hunk 13]
```diff
@@ -7,7 +7,7 @@
     <% end %>
   </div>
   <div>
-    <%= form.label :avatar %><br>
+    <%= form.label :avatar %><br>]
```
Comment (olegsobchuk) [inline app/views/events/_form.html.erb:10]: use `simple_form` gem for this

File: config/routes.rb [hunk 14]
```diff
@@ -9,9 +9,17 @@
   root to: 'pages#home'
   get 'events/index'
   resources :events
+
+  get 'profiles/show'
+  post '/profiles/1/edit'
```
Comment (olegsobchuk) [inline config/routes.rb:14]: WTF???

File: config/routes.rb [hunk 15]
```diff
@@ -9,9 +9,17 @@
   root to: 'pages#home'
   get 'events/index'
   resources :events
+
+  get 'profiles/show'
```
Comment (olegsobchuk) [inline config/routes.rb:13]: use `resources` for this

File: db/migrate/20211116143920_create_profiles.rb [hunk 16]
```diff
@@ -0,0 +1,10 @@
+class CreateProfiles < ActiveRecord::Migration[6.1]
+  def change
+    create_table :profiles do |t|
+      t.string :name
+      t.string :email
```
Comment (olegsobchuk) [inline db/migrate/20211116143920_create_profiles.rb:5]: suppose it should be `required`, use DB validation as well

File: db/migrate/20211116151455_add_user_id_to_profiles.rb [hunk 17]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToProfiles < ActiveRecord::Migration[6.1]
+  def change
+    add_column :profiles, :user_id, :integer
```
Comment (olegsobchuk) [inline db/migrate/20211116151455_add_user_id_to_profiles.rb:3]: it should be `foreign key`

File: app/controllers/profiles_controller.rb (hunk 5 above)
Comment (Nazarii-Makohin) [reply app/controllers/profiles_controller.rb:7]: image
/profiles/show doesn't work then, but /profiles/1 does
Should i link to /profiles/1 then for viewing "My Profile" page?

File: spec/requests/profiles_spec.rb [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+    skip("Add a hash of attributes valid for your model")
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+  describe "GET /index" do
+    it "renders a successful response" do
+      Profile.create! valid_attributes
```
Comment (olegsobchuk) [inline spec/requests/profiles_spec.rb:29]: use `let` 
`it` should consist expectation ONLY. Remove all logic aot of this block

File: spec/requests/profiles_spec.rb [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+      get profiles_url
```
Comment (olegsobchuk) [inline spec/requests/profiles_spec.rb:30]: move to `before`

Comment (olegsobchuk) [reply spec/requests/profiles_spec.rb:30]: don't use `_url` , use `_path`

File: spec/requests/profiles_spec.rb (hunk 18 above)
Comment (olegsobchuk) [reply spec/requests/profiles_spec.rb:29]: the same for all tests

File: spec/routing/profiles_routing_spec.rb [hunk 20]
```diff
@@ -0,0 +1,38 @@
+require "rails_helper"
+
+RSpec.describe ProfilesController, type: :routing do
+  describe "routing" do
+    it "routes to #index" do
+      expect(get: "/profiles").to route_to("profiles#index")
```
Comment (olegsobchuk) [inline spec/routing/profiles_routing_spec.rb:6]: suppose it's pretty clear what you do it these tests. So you can skip description
```
it { expect(get: "/profiles").to route_to("profiles#index") }
```

File: spec/views/profiles/edit.html.erb_spec.rb [hunk 21]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe "profiles/edit", type: :view do
+  before(:each) do
```
Comment (olegsobchuk) [inline spec/views/profiles/edit.html.erb_spec.rb:4]: `before(:each)` => `before` 
`each` is default parameter, so you should skip it

File: spec/views/profiles/edit.html.erb_spec.rb [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+    @profile = assign(:profile, Profile.create!(
```
Comment (olegsobchuk) [inline spec/views/profiles/edit.html.erb_spec.rb:5]: use `FactoryBot` for this

File: config/routes.rb (hunk 14 above)
Comment (Nazarii-Makohin) [reply config/routes.rb:14]: Yeah, i found in a web that i need only resources here, but unfortunately it doesn't work like that without 2 these lines, so Im trying fix it, it's just temporary lines

**Pull request merged**